marklogic-data-hub-central/ui/src/pages/Curate.test.tsx

```diff
@@ -1,9 +1,98 @@
-import React from 'react';
-import { shallow } from 'enzyme';
-import Curate from './Curate';
-
-describe('Install component', () => {
-  it('should render correctly', () => {
-    shallow(<Curate />);
-  });
-});
\ No newline at end of file
+import React, {useContext} from 'react';
+import {render, fireEvent, waitForElement, cleanup} from '@testing-library/react'
+import '@testing-library/jest-dom/extend-expect'
+import {AuthoritiesContext, AuthoritiesService} from '../util/authorities';
+import axiosMock from 'axios';
+import mocks from '../config/mocks.config';
+import Curate from "./Curate";
+
+jest.mock('axios');
+
+describe('Curate component', () => {
+
+    beforeEach(() => {
+        mocks.curateAPI(axiosMock);
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+        cleanup();
+    })
+
+    test('Verify readMapping authority can only view mapping configs and settings', async () => {
+        const authorityService = new AuthoritiesService();
+        authorityService.setAuthorities(['readMapping']);
+
+        const { getByText, getAllByText, queryByText, getByTestId, queryByTestId, debug } = await render(<AuthoritiesContext.Provider value={authorityService}><Curate/></AuthoritiesContext.Provider>);
+
+        expect(await(waitForElement(() => getByText('Customer')))).toBeInTheDocument();
+
+        // Check for steps to be populated
+        expect(axiosMock.get).toBeCalledWith('/api/steps/mapping');
+        fireEvent.click(getByText('Customer'));
+        //Mapping tab should show. Match/Merge should not
+        expect(getByText('Mapping')).toBeInTheDocument();
+        expect(queryByText('Match')).not.toBeInTheDocument();
+        expect(queryByText('Merge')).not.toBeInTheDocument();
+
+        expect(getByText('Mapping1')).toBeInTheDocument();
+
+        // test 'Add New' button
+        expect(queryByText('Add New')).not.toBeInTheDocument();
+
+        // test settings
+        fireEvent.click(getByTestId('Mapping1-settings'));
+        expect(await(waitForElement(() => getByText('Target Database:')))).toBeInTheDocument();
+        expect(getAllByText('Save')[0]).toBeDisabled();
```

Comment:
@bsrikan - Fixed this issue in the PR (https://github.com/bsrikan/marklogic-data-hub/pull/5) against your branch.

Reply:
Thanks @xnikhil08 for the changes. 
As described by Nikhil: "Once you open the settings or edit modal, then it will remain open and when you click on cancel or Save then the parent modal gets a property display: none, 
but the component remains in the document"
Hence we have updated the tests to use data-testid.